### ISSUE TITLE
Variable is not redundant if it have a phpdoc

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#373]
Check the case below (simplified case):

```php
abstract class UserBase
{
}

class User extends UserBase
{
}

function parent(): UserBase
{
    return new User;
}

// Incompatible return type (PS inspection)
function sample1(): User
{
    return parent(); 
}

// Fixing by exporting variable e annotating with User type:
function sample2(): User
{
    /** @var User $user */
    $user = parent();

    return $user;
}
```

It seems to be valid to this inspection, except that `parent()` here returns a `UserBase` that is a `User` parent, not a `User` directly. Then it is marked as "incompatible return type" (that is a PS inspection). To avoid it, we should export the return to a variable and reassign return type with a phpdoc to tell to PhpStorm that we have really a `User` instance. 

I guess that in this case it should be allowed. To be more precise, we should consider to ignore if the variable is annotated with a `@var` property only, not any phpdoc only.

**Comment by kalessil** [comment]
This will do the trick=)

```php
/**
 * @return UserBase|User
 */
function sample2(): User
{
    return parent();
}
```

**Comment by rentalhost** [comment]
Hmm... For some reason it don't worked to me on first try. I will keep investigating it to try identify if it can fail on some specific moment. But for now, not to say. Thanks.

**Comment by kalessil** [comment]
You are welcome =)
